packages/@aws-cdk/aws-autoscaling/lib/auto-scaling-group.ts

```diff
@@ -938,158 +921,34 @@ export interface MetricTargetTrackingProps extends BaseTargetTrackingProps {
-   *
-   * @example '/dev/sdh', 'xvdh'
-   *
-   * @see https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/device_naming.html
-   */
-  readonly deviceName: string;
-
-  /**
-   * Defines the block device volume, to be either an Amazon EBS volume or an ephemeral instance store volume
-   *
-   * @example BlockDeviceVolume.ebs(15), BlockDeviceVolume.ephemeral(0)
-   *
-   */
-  readonly volume: BlockDeviceVolume;
-
-  /**
-   * If false, the device mapping will be suppressed.
-   * If set to false for the root device, the instance might fail the Amazon EC2 health check.
-   * Amazon EC2 Auto Scaling launches a replacement instance if the instance fails the health check.
-   *
-   * @default true - device mapping is left untouched
-   */
-  readonly mappingEnabled?: boolean;
-}
-
-export interface EbsDeviceOptionsBase {
-  /**
-   * Indicates whether to delete the volume when the instance is terminated.
-   *
-   * @default - true for Amazon EC2 Auto Scaling, false otherwise (e.g. EBS)
-   */
-  readonly deleteOnTermination?: boolean;
-
-  /**
-   * The number of I/O operations per second (IOPS) to provision for the volume.
-   *
-   * Must only be set for {@link volumeType}: {@link EbsDeviceVolumeType.IO1}
-   *
-   * The maximum ratio of IOPS to volume size (in GiB) is 50:1, so for 5,000 provisioned IOPS,
-   * you need at least 100 GiB storage on the volume.
-   *
-   * @see https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/EBSVolumeTypes.html
-   */
-  readonly iops?: number;
-
-  /**
-   * The EBS volume type
-   * @see https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/EBSVolumeTypes.html
-   *
-   * @default {@link EbsDeviceVolumeType.GP2}
-   */
-  readonly volumeType?: EbsDeviceVolumeType;
-}
-
-export interface EbsDeviceOptions extends EbsDeviceOptionsBase {
-  /**
-   * Specifies whether the EBS volume is encrypted.
-   * Encrypted EBS volumes can only be attached to instances that support Amazon EBS encryption
-   *
-   * @see https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/EBSEncryption.html#EBSEncryption_supported_instances
-   *
-   * @default false
-   */
-  readonly encrypted?: boolean;
-}
-
-export interface EbsDeviceSnapshotOptions extends EbsDeviceOptionsBase {
-  /**
-   * The volume size, in Gibibytes (GiB)
-   *
-   * If you specify volumeSize, it must be equal or greater than the size of the snapshot.
-   *
-   * @default - The snapshot size
-   */
-  readonly volumeSize?: number;
-}
-
-export interface EbsDeviceProps extends EbsDeviceSnapshotOptions {
-  readonly snapshotId?: string;
-}
-
+export class BlockDeviceVolume extends ec2.BlockDeviceVolume {}
+export interface BlockDevice extends ec2.BlockDevice {}
+export interface EbsDeviceOptions extends ec2.EbsDeviceOptions {}
+export interface EbsDeviceOptionsBase extends ec2.EbsDeviceOptionsBase {}
+export interface EbsDeviceProps extends ec2.EbsDeviceProps {}
+export interface EbsDeviceSnapshotOptions extends ec2.EbsDeviceSnapshotOptions {}
```

Comment:
There's an issue with the interfaces. JSII complains if I attempt to export them:

```
@aws-cdk/aws-autoscaling: error: [awslint:no-unused-type:@aws-cdk/aws-autoscaling.EbsDeviceOptions] type or enum is not used by exported classes (declared at lib/auto-scaling-group.ts:926)
@aws-cdk/aws-autoscaling: error: [awslint:no-unused-type:@aws-cdk/aws-autoscaling.EbsDeviceOptionsBase] type or enum is not used by exported classes (declared at lib/auto-scaling-group.ts:927)
@aws-cdk/aws-autoscaling: error: [awslint:no-unused-type:@aws-cdk/aws-autoscaling.EbsDeviceProps] type or enum is not used by exported classes (declared at lib/auto-scaling-group.ts:928)
@aws-cdk/aws-autoscaling: error: [awslint:no-unused-type:@aws-cdk/aws-autoscaling.EbsDeviceSnapshotOptions] type or enum is not used by exported classes (declared at lib/auto-scaling-group.ts:929)
@aws-cdk/aws-autoscaling: error: [awslint:no-unused-type:@aws-cdk/aws-autoscaling.EbsDeviceVolumeType] type or enum is not used by exported classes (declared at lib/auto-scaling-group.ts:933)
```

Copy-pasta time?

Reply:
Yes. `asg.BlockDevice` doesn't use `asg.EbsDeviceOptions`, it uses `ec2.EbsDeviceOptions`, and so on.

I agree it's not great, but it's practical, and importantly: non-breaking.